test/modules/obsagent-client/src/config.rs

```diff
@@ -0,0 +1,173 @@
+use std::{num::ParseFloatError, string::ParseError};
+
+use clap::{value_t, App, Arg};
+use thiserror::Error;
+
+#[derive(Debug, Clone)]
+pub struct Config {
+    pub update_period: f64,
+    pub prom_config: PromConfig,
+    pub otel_config: OTelConfig,
+}
+
+#[derive(Debug, Clone)]
+pub struct OTelConfig {
+    pub push_period: f64,
+    pub otlp_endpoint: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct PromConfig {
+    pub endpoint: String,
+}
+
+#[derive(Debug, Error)]
+pub enum ConfigError {
+    #[error("error parsing command line argument into string: {0:?}")]
+    StringArgParseError(#[source] ParseError),
+    #[error("error parsing command line argument into float: {0:?}")]
+    FloatArgParseError(#[source] ParseFloatError),
+    #[error("invalid (negative) configuration parameter provided: {0:?}")]
+    InvalidNegativeArgVal(f64),
+    #[error("invalid (zero) configuration parameter provided: {0:?}")]
+    InvalidZeroArgVal(f64),
+}
+
+impl Config {
+    fn new(
+        update_period: f64,
+        prom_config: PromConfig,
+        otel_config: OTelConfig,
+    ) -> Result<Config, ConfigError> {
+        if update_period < 0.0 {
```

Comment:
What if 🐻 puts `f64::INFINITY` here? I know it's silly but I won't trust my future self to use provide this param correctly.

Reply:
Added error handling to make sure f64::INFINITY is not allowed.